### PR TITLE
Add !water command for checking dam levels

### PR DESCRIPTION
**Pull request by JosephGarrone**
Lets you use `!water <REGION>` to check the dam levels of that region. Another regions can be added relatively easily.

### CODE REVIEW

File: uqcsbot/scripts/water.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+from uqcsbot import bot, Command
+import requests
+from urllib.parse import quote
+from bs4 import BeautifulSoup
+from typing import List, Tuple
+from functools import partial
+import asyncio
+from uqcsbot.utils.command_utils import UsageSyntaxException
+
+BASE_URL = "http://acronyms.thefreedictionary.com"
```
Comment (bradleysigma) [inline uqcsbot/scripts/water.py]: Hmmmm.

File: uqcsbot/scripts/water.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+from uqcsbot import bot, Command
+import requests
+from urllib.parse import quote
```
Comment (bradleysigma) [inline uqcsbot/scripts/water.py]: Unused.

File: uqcsbot/scripts/water.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+from uqcsbot import bot, Command
+import requests
+from urllib.parse import quote
+from bs4 import BeautifulSoup
+from typing import List, Tuple
+from functools import partial
```
Comment (bradleysigma) [inline uqcsbot/scripts/water.py]: Unused.

File: uqcsbot/scripts/water.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
+from uqcsbot import bot, Command
+import requests
+from urllib.parse import quote
+from bs4 import BeautifulSoup
+from typing import List, Tuple
+from functools import partial
+import asyncio
```
Comment (bradleysigma) [inline uqcsbot/scripts/water.py]: Unused.

File: uqcsbot/scripts/water.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    response = []
+
+    # Print the list of regions
+    if words[0].lower() in LIST_COMMAND:
+        response.append("Available regions:")
+        for region in REGIONS:
+            response.append(f">{region} (aliases: {', '.join(REGIONS[region]['aliases'])})")
+    else:
+        # Print the info for a specific region
+        if words[0].lower() in REGIONS or words[0].lower() in [alias for region in REGIONS for alias in REGIONS[region]['aliases']]:
```
Comment (bradleysigma) [inline uqcsbot/scripts/water.py]: `if words[0].lower() in REGIONS or words[0].lower() in [alias['aliases'] for alias in REGIONS.values()]`

File: uqcsbot/scripts/water.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                html = BeautifulSoup(http_response.content, 'html.parser')
+
+                maximum_prev_sibling = html.find("div", string='Full supply capacity')
+                maximum_reading = maximum_prev_sibling.find_next_sibling("div")
+                current_prev_sibling = html.find("div", string='Current capacity')
+                current_reading = current_prev_sibling.find_next_sibling("div")
+
+                maximum = int("".join(list(filter(str.isdigit, maximum_reading.get_text()))))
+                current = int("".join(list(filter(str.isdigit, current_reading.get_text()))))
+                percent = (1.0 * current / maximum) * 100.0
```
Comment (bradleysigma) [inline uqcsbot/scripts/water.py]: `100.0 * current / maximum`

Also, `current / maximum` will be a float in Python 3 (`current // maximum` will be an integer).

**Comment by bradleysigma** [comment]
Needs `test_water.py`.

**Comment by JosephGarrone** [comment]
Not sure why build is failing, but doesn't look like my tests are failing. Also noticed that some of the tests seem a bit non-deterministic, got the odd failure on some tests in files I didn't touch when running locally.

**Comment by nicklambourne** [comment]
Appears to be a problem with the call to mypy. I would guess that the invocation pattern has changed, doesn't appear to be a problem with determinism. We'll have to fix the call on our side @kumar-ish.

**Comment by TRManderson** [comment]
From Jenkins: `uqcsbot/scripts/water.py:49: error: Argument 1 has incompatible type "Sequence[str]"; expected "Union[str, bytes]"`

**Comment by nicklambourne** [comment]
jenkins retest this please

**Comment by nicklambourne** [comment]
It appears the scraping is now out of date and is throwing errors. @JosephGarrone happy to reopen this in the future if you feel like looking at it again.

**Pull request closed without merging**